docs/examples/text_classification.md

```diff
@@ -1,31 +1,163 @@
+
+We'll be using AG's news topic classification dataset, a common benchmark dataset for text classification. This dataset
+is a subset of the full AG news dataset, constructed by choosing the four largest classes from the original corpus. Each
+class contains 30,000 training samples and 1,900 testing samples. The total number of training samples is 120,000 with
+7,600 total testing samples.
+
+This dataset contains four columns:
+
+| column      | description                                                                    |
+| ----------- | ------------------------------------------------------------------------------ |
+| class_index | An integer from 1 to 4: "world", "sports", "business", "sci_tech" respectively |
+| class       | A string, one of "world", "sports", "business", "sci_tech"                     |
+| title       | Title of the news article                                                      |
+| description | Description of the news article                                                |
+
+Ludwig also provides several other text classification benchmark datasets which can be used, including:
+
+- [Amazon Reviews](https://s3.amazonaws.com/amazon-reviews-pds/readme.html)
+- [BBC News](https://www.kaggle.com/competitions/learn-ai-bbc/overview)
+- [IMDB](https://www.kaggle.com/datasets/lakshmi25npathi/imdb-dataset-of-50k-movie-reviews)
+- [Yelp Reviews](https://www.kaggle.com/datasets/yelp-dataset/yelp-dataset)
+
+## Download Dataset
+
+Downloads the AG news dataset to the current working directory.
+
+```bash
+ludwig datasets download agnews
 ```
-ludwig experiment \
-  --dataset text_classification.csv \
-  --config config.yaml
+
+This command will download the dataset and write to `agnews.csv` in the current directory.
+
+The CSV file contains the above four columns plus an additional `split` column which is one of 0: train, 1: test,
+2: validation.
+
+Sample (description text omitted for space):
+
+```
+class_index,title,description,split,class
+3,Carlyle Looks Toward Commercial Aerospace (Reuters),...,0,business
+3,Oil and Economy Cloud Stocks' Outlook (Reuters),...,0,business
+3,Iraq Halts Oil Exports from Main Southern Pipeline (Reuters),...,0,business
 ```
 
+## Train
+
+### Define ludwig config
+
+The Ludwig config declares the machine learning task. It tells Ludwig what to predict, what columns to use as input, and
+optionally specifies the model type and hyperparameters.
+
+Here, for simplicity, we'll try to predict **class** from **title**.
+
 With `config.yaml`:
 
 ```yaml
 input_features:
     -
-        name: text
+        name: title
         type: text
-        level: word
         encoder: parallel_cnn
-
 output_features:
     -
         name: class
         type: category
+preprocessing:
+    force_split: true
+    split_probabilities: [0.7, 0.1, 0.2]
```

Comment:
if the dataset already contains the split why are we forcing a different split?

Reply:
This dataset doesn't have a val split (only train and test), and ludwig visualize will error out if there are no validation stats.

Later on I call `ludwig evaluate` with `--split test`, and I realize it is ambiguous which split this is using.
I was hoping nobody would notice this, but since you did it should definitely be fixed :-).

Is there a way to tell Ludwig to auto-split some portion of the training set to use as the val set for training?  Or should I add a val split in the dataset preprocessing?

Reply:
There's no way to do the split of training into training+split right now, but it's definitely a feature we need to add, let's add a task in the project page for it. I'd say in the mean time you can figure out what the best solution is ;)

Reply:
https://github.com/ludwig-ai/ludwig/pull/1868